tests/integrations/raftstore/test_custom_quorum.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::time::Duration;
+
+use raftstore::store::QuorumAlgorithm;
+use test_raftstore::*;
+
+// Test QuorumAlgorithm::IntegrationOnHalfFail works as expected.
+#[test]
+fn test_integration_on_half_fail_quorum_fn() {
+    let mut cluster = new_node_cluster(0, 5);
+    cluster.cfg.raft_store.quorum_algorithm = QuorumAlgorithm::IntegrationOnHalfFail;
+    cluster.run();
+    cluster.must_put(b"k1", b"v0");
+
+    // After peer 4 and 5 fail, no new leader could be elected.
+    cluster.stop_node(4);
+    cluster.stop_node(5);
+    let epoch = cluster.get_region_epoch(1);
+    let cmd = new_admin_request(1, &epoch, new_transfer_leader_cmd(new_peer(2, 2)));
```

Comment:
Why need to transfer leader?

Reply:
Speed up the case, otherwise we need to wait an election timeout to detect no leader.